### ISSUE TITLE
Convox rack password exposed in stack parameters

### ISSUE THREAD

**Issue by mwarkentin** [convox/rack#425]
While poking around in AWS, I was a bit surprised to see the rack password exposed in the CloudFormation parameters list:

![](https://mwarkentin-snaps.s3.amazonaws.com/CloudFormation_Management_Console_2016-03-04_17-20-24.png)

According to [the docs](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html), this parameter could be marked as `NoEcho` to prevent the password from being displayed:

> Important
For sensitive parameter values (such as passwords), set the NoEcho property to true. That way, whenever anyone describes your stack, the parameter value is shown as asterisks (*****).

After discussing w/ @ddollar, he mentioned that this was the current "password recovery" flow for users. Another option might be to follow the standard playbook for website passwords - never "recover", only set a new one. Not sure if there are any other implications to think about.

Slack logs:

```

mwarkentin [5:26 PM] 
Gotcha - thoughts on the Cloudformation Param? Should creds in templates be marked as NoEcho?

[5:27] 
Or is that how the password gets pulled out for the Grid UI, etc.

ddollar [5:28 PM] 
Maybe noecho will have to think about that. Those Parameters are how people recover the password now since its randomly generated

[5:28] 
You can also lock down the cloudformation describe stacks api so you can't read that

mwarkentin [5:30 PM] 
Cool - maybe instead of recovering the password you just set a new one (if you have access to do that)? Similar to a normal web password recovery flow? I don’t know what the implications of that might be. :simple_smile:

ddollar [5:31 PM] 
Definitely worth considering. If you wouldn't mind please file an issue on convox/rack so we can discuss and track
```

**Comment by nzoschke** [comment]
+1 for never recover, always set a new one.

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;tlpghS9X&#x2F;385-plain-text-rack-password"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> plain text rack password</a>

**Comment by ddollar** [comment]
Fixed in #969
